src/core/stream_send.c

```diff
@@ -249,6 +240,10 @@ QuicStreamSendShutdown(
         }
         Stream->Flags.LocalCloseResetReliable = TRUE;
         Stream->SendShutdownErrorCode = ErrorCode;
+
+        // Enqueue all the stuff in ApiSendRequests to SendRequests
+        // It is guaranteed enough sends were queued prior to calling Reliable Shutdown.
```

Comment:
This is still a "todo" right?

Reply:
Correct, there is a lot of logic in SendFlush that may or may not be relevant in this case. It may be worthwhile to leave that for a future PR, and get this PR merged first to address the CLEAR_ALL_SEND_FLAG and stack/heap issue.

Reply:
https://github.com/microsoft/msquic/pull/3900